crucible-llvm/src/Lang/Crucible/LLVM/Translation.hs

```diff
@@ -386,19 +418,21 @@ transDefine halloc ctx d = do
 -- translateModule
 
 -- | Translate a module into Crucible control-flow graphs.
+-- Return the translated module and a list of warning messages
+-- generated during translation.
 -- Note: We may want to add a map from symbols to existing function handles
 -- if we want to support dynamic loading.
 translateModule :: (?transOpts :: TranslationOptions)
                 => HandleAllocator -- ^ Generator for nonces.
                 -> GlobalVar Mem   -- ^ Memory model to associate with this context
                 -> L.Module        -- ^ Module to translate
-                -> IO (Some ModuleTranslation)
+                -> IO (Some ModuleTranslation, [(L.Symbol, Position, Text.Text)])
```

Comment:
There's a lot of tuples in this return type. I wonder if it would be worth wrapping the `(L.Symbol, Position, Text.Text)` part in a newtype.

Reply:
Yeah, that's probably a good idea.